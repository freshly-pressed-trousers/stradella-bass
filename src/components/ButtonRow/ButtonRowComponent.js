import React from "react";
import styles from "./ButtonRowComponent.module.css";

export default function ButtonRowComponent(props) {
	const { children, label } = props;

	return (
		<div className={styles.wrapper}>
			<span className={styles.label}>{label}</span>
			<div className={styles.rowContainer}>{children}</div>
		</div>
	);
}
