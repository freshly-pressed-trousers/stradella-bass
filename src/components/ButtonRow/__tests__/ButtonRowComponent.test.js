import React from "react";
import ButtonRowComponent from "../ButtonRowComponent";
import { shallow } from "enzyme";

describe("ButtonRow component", () => {
	it("renders label", () => {
		const component = shallow(<ButtonRowComponent label="Counter bass" />);

		expect(component.text()).toEqual("Counter bass");
	});

	it("renders children", () => {
		const component = shallow(
			<ButtonRowComponent label="Counter bass">
				<h2>I should be rendered</h2>
			</ButtonRowComponent>
		);

		expect(component.find("h2").exists()).toBe(true);
	});

	it("matches snapshot", () => {
		const component = shallow(<ButtonRowComponent label="Counter bass" />);

		expect(component).toMatchSnapshot();
	});
});
