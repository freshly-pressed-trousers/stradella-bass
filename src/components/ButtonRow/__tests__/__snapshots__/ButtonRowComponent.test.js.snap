// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonRow component matches snapshot 1`] = `
<div
  className="wrapper"
>
  <span
    className="label"
  >
    Counter bass
  </span>
  <div
    className="rowContainer"
  />
</div>
`;
